silkworm/rpc/commands/engine_api_test.cpp

```diff
@@ -198,8 +202,9 @@ TEST_CASE("handle_engine_get_payload_v1 succeeds if request is expected payload"
     ClientContextPool cp{1};
     cp.start();
     std::unique_ptr<ethdb::Database> database;
+    std::unique_ptr<ethbackend::BackEnd> backend;
     // Initialise components
-    EngineRpcApi_ForTest rpc(database.get(), backend.get());
+    EngineRpcApi_ForTest rpc(database.get(), engine.get(), backend.get());
```

Comment:
aren't those NULL here? (and in some other cases in this file)

consider moving those mocks to be created and owned by the EngineRpcApi_ForTest - this would save some repetition in this file

Reply:
I'm currently doing this suggested change and even more refactoring to these tests right now